plugins/modules/zabbix_user_role.py

```diff
@@ -0,0 +1,183 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2022, mrvanes
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+
+DOCUMENTATION = r'''
+---
+module: zabbix_role
+https://www.zabbix.com/documentation/current/en/manual/api/reference/role
+https://www.zabbix.com/documentation/current/en/manual/api/reference/role/create
+
+short_description: Adds or removes zabbix roles
+
+description: This module adds or removes zabbix roles
+
+options:
+    server_url: http://localhost/zabbix
+    login_user: username
+    login_password: password
+    state: exact
+    name: Operators
+      The name of the role
+    type: 1
+      https://www.zabbix.com/documentation/current/en/manual/api/reference/role/object#role
+    rules:
+      https://www.zabbix.com/documentation/current/en/manual/api/reference/role/object#role-rules
```

Comment:
This needs to be valid YAML, it will not parse and pipelines will fail. Links are redundant here, you may add them under `description:` key, which can be either list of strings or just a string

Reply:
also we do not allow referencing Zabbix API that much and try to keep documentation clear of the links that require users to understand and parse API documentation. So this needs to be rewritten here.

For `type: 1` it is preferred to specify string notation and let module handle the translation. In this case there would be `choices ["Admin", "User", "etc..."]` and the module would translate them like this: https://github.com/ansible-collections/community.zabbix/blob/10a4040a8a88d50068d3ecc8475f877db8603864/plugins/modules/zabbix_autoregister.py#L143 (the helper function returns and integer based on the position in the list)

same applies for whole rules section. Here is more complicated example of the module:
https://github.com/ansible-collections/community.zabbix/blob/main/plugins/modules/zabbix_mediatype.py

see how it handles `suboptions` in the DOCUMENTATION block and in the main() function, when defining `argument_spec`